### DISCUSSION TITLE
Hey Spatie Team, I think team there should be an option to clear the complete log when clearing! INDIA :)

### DISCUSSION THREAD

**Discussion by lovekrissh143** [spatie/laravel-ray#98] [Ideas]
I was doing the normal debugging as usual but when I cleared the log, it moved to a new screen with the new log and the last log was persisted which kinda annoyed me first then I co-oped with it. I'm just much much more than happy to get a tool that can help improve my day while I'm climbing my way up. Freeking Great!  :)

But please have an option to clear complete log just like we start from the very clear screen while doing dd();

Appreciate your precious time. :)

**Comment by freekmurze** [comment]
Hi, you can do this with `ray()->newScreen()`

**Reply by lovekrissh143** [reply]
Yeah I read the docs @freekmurze , But what I'm looking for is a way of wiping out the complete dump history.

But when dumping data it keeps track of the earlier screens I've used for dumping which is good in sense of timestamps of my dump work.

But optionally for some users who would want to clear all and have a clean start you can provide that, right @freekmurze .

Currently I've the only way to achieve the same result is by exiting the app and reopening it.

I hope this time I'll be able to make you understand my feature request.

Peace there :)

**Reply by freekmurze** [reply]
In Ray 1.4.0 which will be released shortly, you'll be able to do this. In the latest version of the package you'll be able to call `ray()->clearAll()` to clear everything, including history.

**Reply by lovekrissh143** [reply]
Amazing @freekmurze . Just keep your lovely work going on. This tool is really helpful and this is yet to untap it's full potential in the coming times. :)

**Reply by lovekrissh143** [reply]
Hey @freekmurze I also thought about a situation where I could save all of my debugging history to file of some sort which could be opened later for viewing with the same feel as it's dumped from laravel to keep a little track of what I was working on at that time. 

Just a little bit of memory maintenance for the dump work.

Wouldn't it be a good feature @freekmurze ?

And that ray()->clearAll() would be handy if it's coming as a re-assignable windows shortcut just like we have CTRL+SHIFT+K to open a new screen which I could assign to any shortcut of my liking.

:)

**Reply by lovekrissh143** [reply]
Hey @freekmurze is it possible to get this kinda functionality ?

I'm so excited. :)

**Reply by freekmurze** [reply]
We probably won't add write to a file...

We'll consider adding a clearAll

**Reply by lovekrissh143** [reply]
Great @freekmurze. I'm more than delighted already getting this tool and every new bit of functionality adds a strawberry to the pie.

**Reply by lovekrissh143** [reply]
Hey @freekmurze !

Man I'm getting curious how can I get back the old dump like we used to get in old version of laravel-ray maybe 1.5.0.

Right now when I'm doing ray($admin); it's doing some smart work for me which is in case amazing but what if I want the original details back on the screen.

what was wrong with the older approach ray()->model($user); ?

Right now when using the ray($admin);, the model name, attributes and relations are being picked up automatically which is fantastic but I miss the old dump which was giving me more details about the inner structure which sometimes help in debugging a lot.

![image](https://user-images.githubusercontent.com/48975027/105121915-3b1d5000-5afb-11eb-9bca-311caafbcbc5.png)

Please let me know how can I get back the old details while dumping a model or whatever class not relying on some smart work to figure out on it's own.

I also noticed the change-log saying
 
"the package will now select the best payload type when passing something to ray()" 

which is good but it must preserve some way to dump original details as well which is very much required for debugging. 

I'm missing it Man. Please help!!!
:)

A solution I thought after posting the comment here is that we could get the raw data by doing something like ray($admin)->raw(); or ray()->raw($admin); which I think won't break anything major.

![image](https://user-images.githubusercontent.com/48975027/105132371-d40a9600-5b10-11eb-9851-c8b8d37114dd.png)

This was amazing and providing a lot of info which might not be for everyone but for someone it's a deal.
:)

**Reply by lovekrissh143** [reply]
Hey @freekmurze !

The relations nested inside are showing ellipses...

![image](https://user-images.githubusercontent.com/48975027/105134345-4df04e80-5b14-11eb-8c21-23697e30646c.png)

do I need to apply this kinda hack which I did while doing dd();?

AppServiceProvider.php

```
<?php
use Symfony\Component\VarDumper\VarDumper;
use Symfony\Component\VarDumper\Cloner\VarCloner;
use Symfony\Component\VarDumper\Dumper\CliDumper;
use Symfony\Component\VarDumper\Dumper\HtmlDumper;

VarDumper::setHandler(function ($var) {
    $cloner = new VarCloner();
    $cloner->setMaxItems(-1); // Specifying -1 removes the limit
    $dumper = 'cli' === PHP_SAPI ? new CliDumper() : new HtmlDumper();

    $dumper->dump($cloner->cloneVar($var));
}); 
```

**Reply by freekmurze** [reply]
You can still use `ray()->raw($model)` to get all the details. In our config file there's also a new option to revert to the old behaviour. Check our docs for that.

**Reply by lovekrissh143** [reply]
Wowwwwwwwwww! I tried looking, but I don't know why i didn't get it. Thanks @freekmurze.

**Reply by lovekrissh143** [reply]
> Hey @freekmurze !
> 
> The relations nested inside are showing ellipses...
> 
> ![image](https://user-images.githubusercontent.com/48975027/105134345-4df04e80-5b14-11eb-8c21-23697e30646c.png)
> 
> do I need to apply this kinda hack which I did while doing dd();?
> 
> AppServiceProvider.php
> 
> ```
> <?php
> use Symfony\Component\VarDumper\VarDumper;
> use Symfony\Component\VarDumper\Cloner\VarCloner;
> use Symfony\Component\VarDumper\Dumper\CliDumper;
> use Symfony\Component\VarDumper\Dumper\HtmlDumper;
> 
> VarDumper::setHandler(function ($var) {
>     $cloner = new VarCloner();
>     $cloner->setMaxItems(-1); // Specifying -1 removes the limit
>     $dumper = 'cli' === PHP_SAPI ? new CliDumper() : new HtmlDumper();
> 
>     $dumper->dump($cloner->cloneVar($var));
> }); 
> ```

@freekmurze, I hope you've seen this one.

**Reply by lovekrissh143** [reply]
> Hey @freekmurze !
> 
> The relations nested inside are showing ellipses...
> 
> ![image](https://user-images.githubusercontent.com/48975027/105134345-4df04e80-5b14-11eb-8c21-23697e30646c.png)
> 
> do I need to apply this kinda hack which I did while doing dd();?
> 
> AppServiceProvider.php
> 
> ```
> <?php
> use Symfony\Component\VarDumper\VarDumper;
> use Symfony\Component\VarDumper\Cloner\VarCloner;
> use Symfony\Component\VarDumper\Dumper\CliDumper;
> use Symfony\Component\VarDumper\Dumper\HtmlDumper;
> 
> VarDumper::setHandler(function ($var) {
>     $cloner = new VarCloner();
>     $cloner->setMaxItems(-1); // Specifying -1 removes the limit
>     $dumper = 'cli' === PHP_SAPI ? new CliDumper() : new HtmlDumper();
> 
>     $dumper->dump($cloner->cloneVar($var));
> }); 
> ```

Hey @freekmurze I'm facing this issue again, please help!

**Comment by fieu** [comment]
Why "INDIA" in the title?

**Reply by lovekrissh143** [reply]
@fieu 
Cuz I'm from INDIA and Spatie Team is so generous and superb for INDIANS especially .😂😂👌❤️😍

**Reply by fieu** [reply]
uhm ok
